logins-sql/src/db.rs

```diff
@@ -0,0 +1,844 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use rusqlite::{Connection, types::{ToSql, FromSql}, Row};
+use std::time::SystemTime;
+use std::path::Path;
+use std::collections::HashSet;
+use error::*;
+use schema;
+use login::{LocalLogin, MirrorLogin, Login, SyncStatus, SyncLoginData};
+use sync::{self, ServerTimestamp, IncomingChangeset, Store, OutgoingChangeset, Payload};
+use update_plan::UpdatePlan;
+use util;
+
+pub const MAX_VARIABLE_NUMBER: usize = 999;
+
+pub struct LoginDb {
+    pub db: Connection,
+}
+
+// In PRAGMA foo='bar', `'bar'` must be a constant string (it cannot be a
+// bound parameter), so we need to escape manually. According to
+// https://www.sqlite.org/faq.html, the only character that must be escaped is
+// the single quote, which is escaped by placing two single quotes in a row.
+fn escape_string_for_pragma(s: &str) -> String {
+    s.replace("'", "''")
+}
+
+impl LoginDb {
```

Comment:
I don't find most of this `impl` block to add much value, but as the implementor you get to make that call.  However, I do want to know what our re-use story is: how does the implementor of, say, the history API that's up next take advantage of this layer?

Reply:
A lot of it adds value for debugging -- logging the busted sql statement helps a decent amount IMO. That said it's tricky to use everywhere so I'll see if there's a reason to keep it around I'm not thinking of.

I don't have a plan for history reusing it, beyond thinking that it might be nice.